frontend/src/queries/nodes/DataVisualization/dataVisualizationLogic.ts

```diff
@@ -0,0 +1,283 @@
+    path(['queries', 'nodes', 'DataVisualization', 'dataVisualizationLogic']),
+    connect({
+        values: [teamLogic, ['currentTeamId'], insightSceneLogic, ['insightMode']],
+    }),
+    props({ query: {} } as DataVisualizationLogicProps),
+    propsChanged(({ actions, props }, oldProps) => {
+        if (!props.query) {
+            return // Can't do anything without a query
+        }
+        if (oldProps.query && props.query.kind !== oldProps.query.kind) {
+            actions.clearResponse()
+        }
+        if (!objectsEqual(props.query, oldProps.query)) {
+            if (!props.cachedResults || (!props.cachedResults['result'] && !props.cachedResults['results'])) {
+                actions.loadData()
+            } else {
+                actions.setResponse(props.cachedResults)
+            }
+        }
+    }),
+    actions({
+        loadData: (refresh = false) => ({ refresh, queryId: uuid() }),
+        setResponse: (response: Exclude<AnyResponseType, undefined>) => response,
+        clearResponse: true,
+        abortQuery: (payload: { queryId: string }) => payload,
+        cancelQuery: true,
+        setElapsedTime: (elapsedTime: number) => ({ elapsedTime }),
+        setVisualizationType: (visualizationType: ChartDisplayType) => ({ visualizationType }),
+        setXAxis: (columnIndex: number) => ({ selectedXAxisColumnIndex: columnIndex }),
+        setYAxis: (columnIndex: number) => ({ selectedYAxisColumnIndex: columnIndex }),
+        clearAxis: true,
+        setQuery: (node: DataVisualizationNode) => ({ node }),
+    }),
+    loaders(({ props, actions, cache }) => ({
+        response: [
+            props.cachedResults ?? null,
+            {
+                setResponse: (response) => response,
+                clearResponse: () => null,
+                loadData: async ({ refresh, queryId }, breakpoint) => {
+                    const now = performance.now()
+                    try {
+                        cache.abortController = new AbortController()
+                        const data =
+                            (await query<DataVisualizationNode>(
+                                props.query,
+                                { signal: cache.abortController.signal },
+                                refresh,
+                                queryId
+                            )) ?? null
+                        breakpoint()
+                        actions.setElapsedTime(performance.now() - now)
+                        return data
+                    } catch (e: any) {
+                        actions.setElapsedTime(performance.now() - now)
+                        if (shouldCancelQuery(e)) {
+                            actions.abortQuery({ queryId })
+                        }
+                        breakpoint()
+                        e.queryId = queryId
+                        throw e
+                    }
+                },
+            },
+        ],
+    })),
```

Comment:
Instead of loading the response here, let's just get it from `dataNodeLogic`? This way both the `DataTable` and the `DataVisualization` nodes could share the same query's results.